aibolit/patterns/empty_rethrow/empty_rethrow.py

```diff
@@ -19,29 +19,28 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
-import javalang
-
+from aibolit.ast_framework import ASTNodeType, AST
 from aibolit.utils.ast_builder import build_ast
+from typing import List, Set
 
 
 class EmptyRethrow:
+    '''
+    Check if we throw the same exception as it was caught
+    '''
 
-    def __init__(self):
-        pass
+    def value(self, filename) -> List[int]:
+        total_code_lines: Set[int] = set()
+        ast = AST.build_from_javalang(build_ast(filename))
+        for try_node in ast.get_proxy_nodes(ASTNodeType.TRY_STATEMENT):
+            for throw_node in ast.get_proxy_nodes(ASTNodeType.THROW_STATEMENT):
+                field_catche = try_node.catches
+                if field_catche:
+                    catch_classes = [x.parameter.name for x in field_catche]
+                    throw_child = list(throw_node.children)[0]
+                    if throw_child.node_type == ASTNodeType.CLASS_CREATOR:
+                        continue
+                    if hasattr(throw_child, 'member') and throw_child.member in catch_classes:
+                        total_code_lines.add(throw_child.line)
 
-    def value(self, filename):
-        tree = build_ast(filename)
-        total_code_lines = set()
-        for _, method_node in tree.filter(javalang.tree.MethodDeclaration):
-            for _, try_node in method_node.filter(javalang.tree.TryStatement):
-                for _, throw_node in try_node.filter(javalang.tree.ThrowStatement):
-                    if try_node.catches:
-                        catch_classes = [x.parameter.name for x in try_node.catches]
-                        mem_ref = throw_node.children[1]
-                        if isinstance(mem_ref, javalang.tree.ClassCreator):
-                            continue
-                        else:
-                            if hasattr(mem_ref, 'member') and mem_ref.member in catch_classes:
-                                total_code_lines.add(mem_ref.position.line)
-        return sorted(total_code_lines)
+        return sorted(list(total_code_lines))
```

Comment:
I belive, if we would use that `.subtree` property, there would be no duplicated at all and we could use just list here.